File: basics/06-anagrams/Solution.java
import java.util.Arrays;

class Solution {
    public boolean areAnagrams(String s1, String s2) {
        if (s1.length() != s2.length()) return false;
        char[] a = s1.toLowerCase().toCharArray();
        char[] b = s2.toLowerCase().toCharArray();
        Arrays.sort(a);
        Arrays.sort(b);
        return Arrays.equals(a, b);
    }
}

File: basics/06-anagrams/Solution.cpp
#include <string>
#include <algorithm>

bool areAnagrams(std::string s1, std::string s2) {
    if (s1.size() != s2.size()) return false;
    std::transform(s1.begin(), s1.end(), s1.begin(), ::tolower);
    std::transform(s2.begin(), s2.end(), s2.begin(), ::tolower);
    std::sort(s1.begin(), s1.end());
    std::sort(s2.begin(), s2.end());
    return s1 == s2;
}
